### PR TITLE
Add newline after shebang.

### PR DESCRIPTION
**Pull request by demoneaux**
Not too sure if this formatting is fine as a default, but personally I prefer it this way.

### CODE REVIEW

**Comment by vjeux** [comment]
The stance we've adopted on prettier is to respect newlines that were inside of the original file instead of trying to enforce them in certain places as it is very unlikely we are going to get it right.

The correct fix is to check if apply the newline detection logic to the beginning of the file and respect it.

```js
#!/usr/bin/env node
function f() {}
```

```js
#!/usr/bin/env node

function f() {}
```

should both be re-printed as they are written.

File: index.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ function formatWithShebang(text, opts) {
   const index = text.indexOf("\n");
   const shebang = text.slice(0, index + 1);
   const programText = text.slice(index + 1);
-  return shebang + format(programText, opts);
+  return shebang + "\n" + format(programText, opts);
```
Comment (jlongster) [inline index.js]: @vjeux is right. We should follow the blank line rule like we do elsewhere. Here you can check if `index + 1` is a newline, and if so, add a newline (don't forget, for windows support the check must be `char === "\n" || char === "\r"`.

Comment (demoneaux) [inline index.js]: :+1:, I'll add a check soon!

**Comment by jlongster** [comment]
@demoneaux Any chance you could add a check? I can follow-up tomorrow with it if not.

**Comment by demoneaux** [comment]
I've updated the code to check that case, should work fine now! Note: test failures don't seem related to my change.

**Comment by jlongster** [comment]
I'm considering trying CircleCI. I think the Travis is not giving us the full content of some files and that's why it's been flaky.

**Comment by jlongster** [comment]
We can remove this check if we ever have a more proper fix for keeping the spaces at the top of a file in general. Thanks!

**Pull request merged**